actionpack/test/dispatch/system_testing/driver_test.rb

```diff
@@ -37,6 +49,18 @@ class DriverTest < ActiveSupport::TestCase
     assert_equal ({ url: "http://example.com/wd/hub" }), driver.instance_variable_get(:@options)
   end
 
+  test "initializing the driver with a headless firefox and custom path" do
+    driver_bak = ::Selenium::WebDriver::Firefox::Service.driver_path
+    ::Selenium::WebDriver::Firefox::Service.driver_path = "bin/test"
+    driver = ActionDispatch::SystemTesting::Driver.new(:selenium, using: :headless_firefox, screen_size: [1400, 1400], options: { url: "http://example.com/wd/hub" })
+    assert_equal :selenium, driver.instance_variable_get(:@driver_type)
+    assert_equal :headless_firefox, driver.instance_variable_get(:@browser).name
+    assert_instance_of Selenium::WebDriver::Firefox::Options, driver.instance_variable_get(:@browser).options
+    assert_equal [1400, 1400], driver.instance_variable_get(:@screen_size)
+    assert_equal ({ url: "http://example.com/wd/hub" }), driver.instance_variable_get(:@options)
+    ::Selenium::WebDriver::Firefox::Service.driver_path = driver_bak
```

Comment:
I don't think we need to test so much, we just need to test that a path works. The reset for the driver path also needs to be in an ensure block otherwise it won't get set back if the test fails.

Here's a similar test you can copy https://github.com/rails/rails/blob/a5077bb23e8b2bac84702b43152210518629ed9b/actionpack/test/dispatch/system_testing/driver_test.rb#L151-L162.

Reply:
done. thanks for the help